backend/src/common/client/api_server/v2/experiment_client.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2018-2023 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api_server_v2
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	apiclient "github.com/kubeflow/pipelines/backend/api/v2beta1/go_http_client/experiment_client"
+	params "github.com/kubeflow/pipelines/backend/api/v2beta1/go_http_client/experiment_client/experiment_service"
+	model "github.com/kubeflow/pipelines/backend/api/v2beta1/go_http_client/experiment_model"
+	"github.com/kubeflow/pipelines/backend/src/common/client/api_server"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+	"golang.org/x/net/context"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	apiServerDefaultTimeout = 35 * time.Second
+)
+
+type ExperimentInterface interface {
+	Create(params *params.CreateExperimentParams) (*model.V2beta1Experiment, error)
+	Get(params *params.GetExperimentParams) (*model.V2beta1Experiment, error)
+	List(params *params.ListExperimentsParams) ([]*model.V2beta1Experiment, int, string, error)
+	ListAll(params *params.ListExperimentsParams, maxResultSize int) ([]*model.V2beta1Experiment, error)
+	Archive(params *params.ArchiveExperimentParams) error
+	Unarchive(params *params.UnarchiveExperimentParams) error
+}
+
+type ExperimentClient struct {
+	apiClient      *apiclient.Experiment
+	authInfoWriter runtime.ClientAuthInfoWriter
```

Comment:
Can we remove it? What is it being used for?

Reply:
Removed. Will revisit whether to add/remove this for other APIs later.